### PR TITLE
Implement inhabit v2

### LINKED ISSUE #105: Implement inhabit subcommand
Switch with hermit shell is currently inhabited.  This means changing the global storage of what the shell's name is as well as removing symlinks to the old shell's files, and creating symlinks to the new shell's files.

### LINKED ISSUE #116: Implement link utility

### LINKED ISSUE #117: Implement unlink utility

### PR DESCRIPTION
**Pull request by RadicalZephyr**
Fixes #105, fixes #116, fixes #117 

### Summary

The primary code addition here is an extension to the `Config` trait.  We add a method `shell_files` which returns an iterator (sort of) over all the files in the shell.

The reasoning behind this is to continue using the `Config` trait as the abstraction break for filesystem access of hermit configuration.

Implementing this method and designing the trait interface correctly took quite a while, and a lot of different approaches were explored.  I think the current solution is pretty clean, and it allows for really trivial mocking of the `shell_files` method by having the `MockConfig` struct return a `Vec<PathBuf>`. This simplicity of mocking is mostly what drove the design decision to have the trait actually return an `IntoIterator` rather than an `Iterator`.  This choice also made it easier to write the "real" filesystem-accessing implementation.

#### Dependency Addition

A note on the dependency that is added: [walkdir](https://github.com/BurntSushi/walkdir).

This crate looks to be pretty robust, and has all the options we need.  It's also written by a fairly prominent Rustacean (BurntSushi) who basically (through his blog posts) taught me much of what I know about Rust, particularly how to do ergonomic and effective error handling.  Furthermore, this crate is the product of him attempting to fix/complete/stabilize the [unstable directory walking implementation in std](https://doc.rust-lang.org/1.5.0/std/fs/fn.walk_dir.html).

TL;DR I think he knows what he's doing :smile:

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -165,63 +273,92 @@ mod test {
     #[test]
     fn has_a_root_path() {
         let test_root = set_up("root-path", "default", vec!["default"]);
-        let config = FsConfig::new(test_root.clone());
+        let config = FsConfig::new(&test_root);
```
Comment (reillysiemens) [inline src/config.rs]: :heavy_plus_sign::100:
